### PR TITLE
Try: [Block] Credits block

### PR DESCRIPTION
**Pull request by MaggieCabrera**
## Description

This PR creates a credits block that inserts outputs a translatable string selected from some options on the sidebar. Addresses https://github.com/WordPress/gutenberg/issues/33475 

## How has this been tested?

Tested on emptytheme

## Types of changes

New feature (non-breaking change which adds functionality) 

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by scruffian** [comment]
It would be nice to add the option to display an icon, and maybe have an "icon only" option too...

**Comment by scruffian** [comment]
I wonder if we should remove this from the inserter?

File: packages/block-library/src/credits/edit.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	attributes: { contentID, content, textAlign },
+	setAttributes,
+} ) {
+	const blockProps = useBlockProps( {
+		className: classnames( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		} ),
+	} );
+	const creditOptions = [
+		{ value: 0, label: __( 'Built with WordPress' ) },
```
Comment (scruffian) [inline packages/block-library/src/credits/edit.js:29]: Do we need the value key, or can we just rely on the position in the array?

File: packages/block-library/src/credits/edit.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+						setAttributes( { textAlign: nextAlign } );
+					} }
+				/>
+			</BlockControls>
+			<InspectorControls key="setting">
+				<PanelBody title={ __( 'Settings' ) }>
+					<SelectControl
+						options={ creditOptions }
+						value={ contentID }
+						label={ __( 'Credits text' ) }
```
Comment (scruffian) [inline packages/block-library/src/credits/edit.js:65]: I think this should probably say "Credit text"

File: packages/block-library/src/credits/block.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+			"default": 0
+		},
+		"content": {
+			"type": "string",
+			"default": ""
+		},
+		"textAlign": {
+			"type": "string"
+		}
+	},
```
Comment (scruffian) [inline packages/block-library/src/credits/block.json:20]: If we do decide to show it in the inserter we might want to add some keywords:
`credit` and `wordpress` would make sense.

File: packages/block-library/src/credits/edit.js (hunk 1 above)
Comment (MaggieCabrera) [reply packages/block-library/src/credits/edit.js:29]: we need the value key for the select, sadly :(

File: packages/block-library/src/credits/edit.js (hunk 2 above)
Comment (scruffian) [reply packages/block-library/src/credits/edit.js:65]: I actually think the whole block should be "credit" not "credits".

**Pull request closed without merging**